### ISSUE TITLE
The gradient through `split' function maybe a little wrong

### ISSUE THREAD

**Issue by XgDuan** [pytorch/pytorch#37827]
## 🐛 Bug

The gradient backpropagated through the `split' function would be wrong, at least in the following scenario:
When the input is a 3 dimension tensor and is split along the second dimension and produced with another tensor using each split. The gradient will be cumulated along the first dimension (which is not supposed to be so.)

## To Reproduce

1. Running the followed code:
```
import torch
a = torch.rand(2, 3, 5, requires_grad=True)
a1, a2, a3 = a.split([1,1,1], dim=1)
weight = torch.rand(2, 5)
loss = (a1 * weight + a2 * weight + a3 * weight).sum()
loss.backward()
```
2. The gradient wrt `a' should be repentances of `weight' alone the first dimension. However, I find that the gradient wrt.  `a' was summed along the first dimension.

![image](https://user-images.githubusercontent.com/13313976/81086514-2d9f4600-8f2b-11ea-855d-ee1bdc46e812.png)

## Expected behavior

The expected result should be the same with the following codes:
```
import torch
a = torch.rand(2, 3, 5, requires_grad=True)
a1, a2, a3 = a[:, 0, :], a[:, 1, :], a[:, 2, :]
weight = torch.rand(2, 5)
loss = (a1 * weight + a2 * weight + a3 * weight).sum()
loss.backward()
```
![image](https://user-images.githubusercontent.com/13313976/81086624-56bfd680-8f2b-11ea-9790-8417b0b85164.png)

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.1, 1.3, 1.4 (All be tested)
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version: 3.6, 3.7
 - CUDA/cuDNN version:  cuda9.0, cuda 10.2
 - GPU models and configuration:
 - Any other relevant information:

**Comment by SsnL** [comment]
The autograd is correct. Your first code snippet does broadcasting...
`a1.shape == [2, 1, 5]`.
When multiplied with `weight` of shape `[2, 5]`, broadcasting gives you a result of shape `[2, 2, 5]`.
